base/compiler/ssair/passes.jl

```diff
@@ -2331,8 +2340,11 @@ function cfg_simplify!(ir::IRCode)
                     end
                     curr = merged_succ[curr]
                 end
-                terminator = ir[SSAValue(ir.cfg.blocks[curr].stmts[end])][:stmt]
-                if isa(terminator, GotoNode) || isa(terminator, ReturnNode)
+                terminator = ir[SSAValue(bbs[curr].stmts[end])][:stmt]
+                is_throw = ir[SSAValue(bbs[curr].stmts[end])][:type] === Union{}
+                if isa(terminator, GotoNode) || isa(terminator, ReturnNode) || is_throw
```

Comment:
There's a corner case for PhiNode where the type is Union{} and the stmt is implicit fall through, but it's not unreachable.

Reply:
Good catch - Thanks for the heads up